File: frontend/src/types/pricingRule.ts
export interface PricingRuleDto {
  id: string;
  screenId: string;
  name: string;
  ruleType: string;
  regularSlotPrice: number | null;
  isActive: boolean;
  startDate: string | null;
  endDate: string | null;
  daysOfWeek: string | null;
  createdAt: string;
}

export interface CreatePricingRuleRequest {
  name: string;
  ruleType: string;
  regularSlotPrice?: number;
  isActive: boolean;
  startDate?: string;
  endDate?: string;
  daysOfWeek?: string;
}

File: backend/CCMS.Domain/Entities/PricingRule.cs
using CCMS.Domain.Enums;

namespace CCMS.Domain.Entities;

public class PricingRule : BaseEntity
{
    public Guid ScreenId { get; set; }
    public string Name { get; set; } = string.Empty;
    public PricingRuleType RuleType { get; set; }

    // Prices (null = use screen base price)
    public decimal? RegularSlotPrice { get; set; }

    public bool IsActive { get; set; } = true;

    // For DateRange and SpecificDate rule types
    public DateOnly? StartDate { get; set; }
    public DateOnly? EndDate { get; set; }

    // For Weekday rule type: comma-separated day numbers (0=Sunday, 1=Monday, ..., 6=Saturday)
    public string? DaysOfWeek { get; set; }

    // Navigation
    public virtual Screen Screen { get; set; } = null!;
}
